---
title: "Small RNA-seq quality control"
author: "`r getOption('author')`"
date: "`r Sys.Date()`"
output:
    html_document:
        code_folding: hide
        df_print: paged
        highlight: pygments
        number_sections: false
        self_contained: true
        theme: paper
        toc: true
        toc_float:
            collapsed: true
            smooth_scroll: false
params:
    bcb: "data/bcb.rda"
    outputDir: "."
---

```{r aes, results='hide'}
library(knitr)
library(ggplot2)

# Set seed for reproducibility
set.seed(1454944673L)

opts_chunk[["set"]](
    autodep = TRUE,
    bootstrap.show.code = FALSE,
    cache = TRUE,
    cache.lazy = TRUE,
    dev = c("png", "pdf"),
    error = TRUE,
    fig.height = 6,
    fig.retina = 2L,
    fig.width = 6,
    highlight = TRUE,
    message = FALSE,
    prompt = TRUE,
    # formatR required for tidy code
    tidy = TRUE,
    warning = FALSE)

theme_set(
    theme_light(base_size = 11L))
theme_update(
    legend.justification = "center",
    legend.position = "bottom")
```

```{r setup, message=FALSE}
library(isomiRs)
library(DEGreport)
library(bcbioSmallRna)
library(ComplexHeatmap)
library(circlize)

out_dir <- params$outputDir
data_out <- file.path(out_dir, "data")
res_out <- file.path(out_dir, "results")
sapply(list(res_out, data_out), dir.create, recursive = TRUE, showWarnings = FALSE)

# bcbioSmallRnaDataSet
bcb_name <- load(params$bcb)
bcb <- get(bcb_name, inherits = FALSE)
```

```{r header, eval = file.exists("_header.Rmd"), child="_header.Rmd"}
```


# Exploratory analysis

In this section we will see descriptive figures about quality of the data, 
reads with adapter, reads mapped to miRNAs, reads mapped to other small RNAs. 


## Size distribution

After adapter removal, we can plot the size distribution of the small RNAs.
We expect the majority of reads to have adapters and we expect a peak
at 22 (maybe 33 as well) that indicates miRNA/tRNA enrichment. There
are some cases this rule won't apply.

```{r adapter, fig.height=7}
bcbSmallSize(bcb)
```

```{r adapter-size-dist, fig.height=7}
bcbSmallSizeDist(bcb, color = "lab")
```

## miRNA

A microRNA (abbreviated miRNA) is a small non-coding RNA molecule (containing about 22 nucleotides) found in plants, animals and some viruses, that functions in RNA silencing and post-transcriptional regulation of gene expression.[https://en.wikipedia.org/wiki/MicroRNA]

### Total miRNA expression annotated with mirbase

[miRBase](www.mirbase.org) is one of the database that contains the reference miRNA sequences. We used this database to annotate the sequences with [miraligner](http://seqcluster.readthedocs.io/mirna_annotation.html)

```{r plot_mirna_counts, fig.height=7, fig.cap="Bottom figure tells how many miRNAs until saturation has been detected. Flatten at low value means few miRNAs capture the majority of the coverage."}
bcbSmallMicro(bcb, color = "lab")
```

### Clustering of samples

```{r mirna-cluster-data}
data = bcbSmallPCA(bcb, columns = c("lab", "protocol"))
color_by = metadata(bcb)[["interesting_groups"]]
```

```{r mirna-heatmap}
palette <- colorRamp2(seq(min(data[["counts"]]),
                          max(data[["counts"]]), length = 3),
                      c("blue", "#EEEEEE", "orange"), space = "RGB")
th <- HeatmapAnnotation(df = data[["annotation"]],
                        col = degColors(data[["annotation"]], palette = "Set1"))
Heatmap(data[["counts"]],
        col = palette,
        top_annotation = th,
        clustering_method_rows = "ward.D",
        clustering_distance_columns = "kendall",
        clustering_method_columns = "ward.D",
        show_row_names = FALSE,
        show_column_names = ncol(data[["counts"]]) < 50)
```

```{r mirna-pca}
degPCA(data[["counts"]], data[["annotation"]],
            condition = "protocol")
```


## isomiRs

isomiR is a term coined by Morin et al. to refer to those sequences that have variations with respect to the reference MiRNA sequence. [https://en.wikipedia.org/wiki/IsomiR]

There are 5 different types of isomiRs:

* ref: perfect match to the sequence on the database
* t3: nucleotides changes at 3'
* t5: nucleotides changes at 5'
* add: nucleotides addition at the 3'
* mism: putatives SNPs

![](http://ars.els-cdn.com/content/image/1-s2.0-S0168952512001126-gr1.jpg)

```{r isomiRs, fig.cap="Proportion of isomiRs counting reads (left) or unique sequences (right)."}
ids <- bcbio(bcb, "isomirs")
isoPlot(ids, type = "all", column = "protocol")
```


## Small RNA clusters

We use [seqcluster](http://seqcluster.readthedocs.io/getting_started.html) to identify any type of small RNA. It generates a list of clusters of small RNA sequences, their genome location, their annotation and the abundance in all the sample of the project. The next figure explain the algorithm used for that and the way to use it for differential expression analyses.

<div style="width:150px; height=150px">
![](http://seqcluster.readthedocs.io/_images/seqcluster.png){ width=50% }
</div>

This file generated by seqcluster, `seqcluster.db`, can be used with the page `reader.html` after downloading from [here](https://github.com/lpantano/seqclusterViz/archive/master.zip). 

See example:

[![IMAGE ALT TEXT](http://i.makeagif.com/media/7-03-2016/M0GjW2.gif)](https://youtu.be/Zjzte8n2-Sg "Example")


```{r plot_srna_clusters, fig.width=6, fig.height=8, fig.cap="Reads kept after each filering step(top left). Proportion of sequences detected (top right). SmallRNA proportions (bottom)."}
bcbSmallCluster(bcb)
```


### Clustering of samples

```{r cluster-cluster-data}
data = bcbSmallPCA(bcb, "cluster")
color_by = metadata(bcb)[["interesting_groups"]]
```

```{r cluster-heatmap}
palette <- colorRamp2(seq(min(data[["counts"]]),
                          max(data[["counts"]]), length = 3),
                      c("blue", "#EEEEEE", "orange"), space = "RGB")
th <- HeatmapAnnotation(df = data[["annotation"]],
                        col = degColors(data[["annotation"]]))
Heatmap(data[["counts"]],
        col = palette,
        top_annotation = th,
        clustering_method_rows = "ward.D",
        clustering_distance_columns = "kendall",
        clustering_method_columns = "ward.D",
        show_row_names = FALSE,
        show_column_names = ncol(data[["counts"]]) < 50)
```

```{r cluster-pca}
degPCA(data[["counts"]], data[["annotation"]],
            condition = color_by)
```


```{r footer, eval = file.exists("_footer.Rmd"), child="_footer.Rmd"}
```
